src/main/scala/cc/factorie/app/nlp/ner/StackedChainNer.scala

```diff
@@ -311,9 +311,13 @@ class StackedChainNer[L<:NerTag](labelDomain: CategoricalDomain[String],
         features += "CLUS="+prefix(20,clusters(rawWord))
       }
     }
-    for (sentence <- document.sentences) cc.factorie.app.chain.Observations.addNeighboringFeatureConjunctions(sentence.tokens, vf, "^[^@]*$", List(0), List(1), List(2), List(-1), List(-2))
-    // Add features for character n-grams between sizes 2 and 5
-    document.tokens.foreach(t => if (t.string.matches("[A-Za-z]+")) vf(t) ++= t.charNGrams(2,5).map(n => "NGRAM="+n))
+    for (sentence <- document.sentences)
+      cc.factorie.app.chain.Observations.addNeighboringFeatureConjunctions(sentence.tokens, vf, FEATURE_PREFIX_REGEX, List(0), List(1), List(2), List(-1), List(-2))
+
+    document.tokens.foreach { t =>
```

Comment:
It didn't look that important during the profiling runs, but I agree it would be cleaner. The ones that are applying a lambda I didn't want to rewrite as it'll be a complicated change (and my lambdas aren't very scala-y so it might be best for someone else to do it).